src/Orleans/Runtime/OutsideRuntimeClient.cs

```diff
@@ -134,6 +134,9 @@ public OutsideRuntimeClient(ClientConfiguration cfg, GrainFactory grainFactory,
             logger = LogManager.GetLogger("OutsideRuntimeClient", LoggerType.Runtime);
             appLogger = LogManager.GetLogger("Application", LoggerType.Application);
 
+            BufferPool.InitGlobalBufferPool(config);
+            this.clientId = GrainId.NewClientId(cfg.ClusterId);
```

Comment:
The call to `OutsideRuntimeClient.StartInternal()` is too late for giving us the clusterId. By then, the client has already initialized lots of other objects passing the clientId (those are not easily updated), and has already established a socket connection.

Given that, I think I would rather stick to the manual configuration as currently implemented. To make it a bit less error-prone, there is a check to catch wrong configuration (generates a fatal error if the user specifies a cluster id in the client config that is different from the cluster id in the silo config).